src/vs/workbench/contrib/positronVariables/browser/components/variablesCore.tsx

```diff
@@ -31,6 +34,56 @@ interface VariablesCoreProps extends PositronVariablesProps {
 export const VariablesCore = (props: VariablesCoreProps) => {
 	// Context hooks.
 	const positronVariablesContext = usePositronVariablesContext();
+	const progressRef = useRef<HTMLDivElement>(null);
+
+	useEffect(() => {
+		const disposables = new DisposableStore();
+
+		let progressBar: ProgressBar | undefined;
+
+		const clearProgressBar = () => {
+			if (progressBar) {
+				progressBar.done();
+				progressBar.dispose();
+				progressBar = undefined;
+				progressRef.current?.replaceChildren();
+			}
+		}
+
+		const setProgressBar = () => {
+			// No work to do if we don't have a progress bar.
+			if (!progressRef.current) {
+				return;
+			}
+
+			// Before starting a new render, remove any existing progress bars. This prevents
+			// a buildup of progress bars when rendering multiple times and ensures the progress bar
+			// is removed when a new render is requested before the previous one completes.
+			progressRef.current.replaceChildren();
+			// Create the progress bar.
+			progressBar = new ProgressBar(progressRef.current);
+			progressBar.infinite();
+		}
+
+		if (positronVariablesContext.activePositronVariablesInstance) {
+			disposables.add(positronVariablesContext.activePositronVariablesInstance.onDidChangeStatus((status) => {
+				if (status === RuntimeClientStatus.Busy) {
+					setProgressBar();
```

Comment:
Should this be slightly debounced? I don't know if we'd want to flicker a progress bar in the average case (e.g. expansion takes < 400ms).

Reply:
Done in https://github.com/posit-dev/positron/pull/7286/commits/4bb14d52a9e83a9975f0538de0b9ea1655152c67